logstash-core/spec/api/lib/api/plugins_spec.rb

```diff
@@ -0,0 +1,45 @@
+# encoding: utf-8
+require_relative "../../spec_helper"
+require "sinatra"
+require "app/modules/plugins"
+require "logstash/json"
+
+describe LogStash::Api::Plugins do
+
+  include Rack::Test::Methods
+
+  def app()
+    described_class
+  end
+
+  before(:all) do
+    get "/"
+  end
+
+  let(:payload) { JSON.parse(last_response.body) }
+
+  it "respond to plugins resource" do
+    expect(last_response).to be_ok
+  end
+
+  it "return valid json content type" do
+    expect(last_response.content_type).to eq("application/json")
+  end
+
+  it "return the expected schema" do
+    expect(payload.keys).to include("plugins", "total")
+    payload["plugins"].each do |plugin|
+      expect(plugin.keys).to include("name", "version")
```

Comment:
(excuse my intrusion) 

`nil` is a bitch, I wish it did not exist :P but I believe that when testing an interface where `nil` values are illegal, we should add a specific test for it. now, as far as testing for proper content format, it's not a bad idea, but pragmatically I thinks that just checking that the return type is `String` and non-empty would be sufficient IMO.